fv3core/pace/fv3core/stencils/c_sw.py

```diff
@@ -346,20 +365,20 @@ def transportdelp_update_vorticity_and_kineticenergy(
 
     with computation(PARALLEL), interval(...):
         # update vorticity and kinetic energy
-        compile_assert(grid_type < 3)
 
         ke = uc if ua > 0.0 else uc[1, 0, 0]
         vort = vc if va > 0.0 else vc[0, 1, 0]
 
-        with horizontal(region[:, j_start - 1], region[:, j_end]):
-            vort = vort * sin_sg4 + u[0, 1, 0] * cos_sg4 if va <= 0.0 else vort
-        with horizontal(region[:, j_start], region[:, j_end + 1]):
-            vort = vort * sin_sg2 + u * cos_sg2 if va > 0.0 else vort
+        if __INLINED(grid_type < 3):
```

Comment:
Moving `grid_type` to `int` might be an issue for this type of code. So we might need a `as_int` method.

That should be doable though - this is the typical feature we need in gt4py to make it comfortable